packages/manager/apps/pci-block-storage/src/pages/new/components/CapacityStep.component.tsx

```diff
@@ -53,21 +53,29 @@ export function CapacityStep({
   const { t: tGlobal } = useTranslation('global');
   const [volumeCapacity, setVolumeCapacity] = useState<number>(VOLUME_MIN_SIZE);
   const [maxSize, setMaxSize] = useState(0);
-  const { data: regionQuotas, isLoading } = useRegionsQuota(projectId);
+  const {
+    data: regionQuotas,
+    isLoading: isRegionQuotaLoading,
+  } = useRegionsQuota(projectId, region.name);
   const isCapacityValid =
     volumeCapacity >= VOLUME_MIN_SIZE && volumeCapacity <= maxSize;
 
+  const { volumeMaxSize, isLoading: isVolumeMaxSizeLoading } = useVolumeMaxSize(
+    region.name,
+  );
+
+  const isLoading = isRegionQuotaLoading && isVolumeMaxSizeLoading;
+
   useEffect(() => {
     if (!isLoading) {
-      const quota = regionQuotas?.find(({ region: r }) => r === region?.name);
-      let availableGigabytes = VOLUME_MAX_SIZE;
+      let availableGigabytes = volumeMaxSize;
       if (
-        quota?.volume &&
-        quota.volume.maxGigabytes !== VOLUME_UNLIMITED_QUOTA
+        regionQuotas?.volume &&
+        regionQuotas.volume.maxGigabytes !== VOLUME_UNLIMITED_QUOTA
       ) {
         availableGigabytes = Math.min(
-          VOLUME_MAX_SIZE,
-          quota.volume.maxGigabytes - quota.volume.usedGigabytes,
+          volumeMaxSize,
+          regionQuotas.volume.maxGigabytes - regionQuotas.volume.usedGigabytes,
```

Comment:
Will be better to move the logic into a dedicated function, what do you think?

Reply:
This will lead to a lot of changes, global page UX should be improved later